### PR TITLE
Fix 16805: pagination dropdown issue

### LINKED ISSUE #16805: Search Results Pagination Issue
**Affected module**
UI

**Describe the bug**
When I search for a specific ingestion, if the search result is on the second page, entering the filter value in the searching box won't show the corresponding result on the first page; I still have to go to the second page to find it. 
**To Reproduce**
Screenshots or steps to reproduce
image

**Expected behavior**
Can see the result on the first page
**Version:**
 - OS: Redhat8.9
 - Python version: 3.9
 - OpenMetadata version: 1.4.1
 - OpenMetadata Ingestion package version: bare metal

**Additional context**
thread in slack:
https://openmetadata.slack.com/archives/C02B6955S4S/p1719382295767539

### PR DESCRIPTION
**Pull request by pranita09**
Fixes #16805

I worked on pagination dropdown for the pages, DatabaseSchemaPage, APICollectionPage, ServiceDetailsPage, etc.

Screenshot 2024-11-12 at 7 53 13 PM

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/pages/APICollectionPage/APICollectionPage.tsx [hunk 1]
```diff
@@ -454,13 +455,13 @@ const APICollectionPage: FunctionComponent = () => {
...
           paging: {
-            [cursorType]: apiEndpoints.paging[cursorType],
+            [cursorType]: paging[cursorType],
           },
         });
       }
       setCurrentEndpointsPage(currentPage);
     },
-    [apiEndpoints, getAPICollectionEndpoints]
+    [paging, getAPICollectionEndpoints, handlePageChange]
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/pages/APICollectionPage/APICollectionPage.tsx:464]: We are not using handlePageChange in the function so i dont think we need as dependancy

File: openmetadata-ui/src/main/resources/ui/src/pages/APICollectionPage/APICollectionVersionPage.tsx [hunk 2]
```diff
@@ -228,13 +231,13 @@ const APICollectionVersionPage = () => {
...
           paging: {
-            [cursorType]: apiEndpoints.paging[cursorType],
+            [cursorType]: paging[cursorType],
           },
         });
       }
       setCurrentEndpointsPage(currentPage);
     },
-    [apiEndpoints, getAPICollectionEndpoints]
+    [paging, getAPICollectionEndpoints, handlePageChange]
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/pages/APICollectionPage/APICollectionVersionPage.tsx:240]: `handlePageChange` is not used in the function

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/DatabaseSchemaPage.component.tsx [hunk 3]
```diff
@@ -452,11 +455,11 @@ const DatabaseSchemaPage: FunctionComponent = () => {
   const tablePaginationHandler = useCallback(
     ({ cursorType, currentPage }: PagingHandlerParams) => {
       if (cursorType) {
-        getSchemaTables({ [cursorType]: tableData.paging[cursorType] });
+        getSchemaTables({ [cursorType]: paging[cursorType] });
       }
       setCurrentTablesPage(currentPage);
     },
-    [tableData, getSchemaTables]
+    [paging, getSchemaTables, handlePageChange]
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/DatabaseSchemaPage.component.tsx:462]: same comment here

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaVersionPage/DatabaseSchemaVersionPage.tsx [hunk 4]
```diff
@@ -189,11 +193,11 @@ function DatabaseSchemaVersionPage() {
   const tablePaginationHandler = useCallback(
     ({ cursorType, currentPage }: PagingHandlerParams) => {
       if (cursorType) {
-        getSchemaTables({ [cursorType]: tableData.paging[cursorType] });
+        getSchemaTables({ [cursorType]: paging[cursorType] });
       }
       setCurrentPage(currentPage);
     },
-    [tableData, getSchemaTables]
+    [paging, getSchemaTables, handlePageChange]
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaVersionPage/DatabaseSchemaVersionPage.tsx:200]: same here

File: openmetadata-ui/src/main/resources/ui/src/pages/APICollectionPage/APICollectionPage.tsx (hunk 1 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/pages/APICollectionPage/APICollectionPage.tsx]: Removed handlePageChange dependency.

File: openmetadata-ui/src/main/resources/ui/src/pages/APICollectionPage/APICollectionVersionPage.tsx (hunk 2 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/pages/APICollectionPage/APICollectionVersionPage.tsx]: Removed the dependency.

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/DatabaseSchemaPage.component.tsx (hunk 3 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/DatabaseSchemaPage.component.tsx]: Removed dependency.

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaVersionPage/DatabaseSchemaVersionPage.tsx (hunk 4 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaVersionPage/DatabaseSchemaVersionPage.tsx]: Removed dependency.

**Review by Sachin-chaurasiya — DISMISSED** [review]

**Review by ShaileshParmar11 — COMMENTED** [review]
need to fix playwright before merging

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**